togetherly/lib/services/reward_redemption_service.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:supabase_flutter/supabase_flutter.dart';
+import 'package:togetherly/models/reward_redemption.dart';
+
+class RewardRedemptionService {
+  static const String _rewardRedemptionTable = "redeemed_reward";
+  static const String _familyRedeemedRewardsView = "family_redeemed_rewards";
+
+  Future<List<RewardRedemption>> getRedeemedRewardsByFamily(
+      int familyId) async {
+    var result = await Supabase.instance.client
+        .from(_familyRedeemedRewardsView)
+        .select('family_id, person_id, reward_id, quantity, timestamp')
+        .eq("family_id", familyId);
+    return result.map(_mapToRedeemedReward).toList();
+  }
+
+  Future<RewardRedemption> insertReward(
+      int familyId, RewardRedemption redemption) async {
```

Comment:
See my comment below on the `_rewardToMap` function. The `familyId` parameter can be removed.

Reply:
So I realized I needed to change the table to the view but don't we need to get them by familyid?

Reply:
The view is what we use to get items by family-id, and the very reason it exists is because the rows don't directly store a family ID. Thus, we need to get rows by family-id, but we don't need a family-id when inserting or updating rows. I see that the familyId parameter is still here, but I can push a change to remove it.